state-chain/pallets/cf-swapping/src/lib.rs

```diff
@@ -826,7 +830,16 @@ pub mod pallet {
 					"All swaps should have Stable amount set here"
 				);
 				let stable_amount = swap.stable_amount.get_or_insert_with(Default::default);
-				*stable_amount = T::SwappingApi::take_network_fee(*stable_amount).remaining_amount;
+
+				let NetworkFeeTaken { remaining_amount, network_fee } =
+					T::SwappingApi::take_network_fee(*stable_amount);
```

Comment:
I was actually thinking we could move the whole implementation into the swapping pallet, including the storage and swapping logic. Keeping it in the pools pallet seems like it adds an unnecessary layer of indirection.

Reply:
But this is a good start 😅

Reply:
Created https://linear.app/chainflip/issue/PRO-1404/move-collectednetworkfee-and-related-logic-into-the-swapping-pallet (which you can feel free to edit). Let me know if this PR can be merged.